torchmetrics/classification/auroc.py

```diff
@@ -118,20 +119,30 @@ def __init__(
         max_fpr: Optional[float] = None,
         compute_on_step: bool = True,
         dist_sync_on_step: bool = False,
+        reduce_memory: bool = False,
         process_group: Optional[Any] = None,
         dist_sync_fn: Callable = None,
+        max_threshold_count = 100,
+        num_batches_to_sample = 4,
```

Comment:
These new arguments should be added to the docstring, to explain what they do. Of particular interest might be how `reduce_memory` reduces memory.